src/hooks/core/chat/useFetchPostNotificationHook.ts

```diff
@@ -2,29 +2,41 @@
 import AnonymousMessageRepo from '../../../service/repo/anonymousMessageRepo';
 import ChannelList from '../../../database/schema/ChannelListSchema';
 import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
+import useDatabaseQueueHook from '../queue/useDatabaseQueueHook';
 import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import StorageUtils from '../../../utils/storage';
 
 const useFetchPostNotificationHook = () => {
   const {localDb, refresh} = useLocalDatabaseHook();
+  const {queue} = useDatabaseQueueHook();
 
   const saveNotifications = async (notifications: any[], fromNotificationAPI: any) => {
     if (!localDb) return;
-    const allPromises: Promise<void>[] = notifications.map((notification) => {
-      const channelList = fromNotificationAPI(notification);
-      return channelList.save(localDb);
+
+    notifications.map((notification) => {
+      queue.addJob({
+        label: 'save post notifications',
+        task: async () => {
+          const channelList = fromNotificationAPI(notification);
+          await channelList.save(localDb);
+        }
+      });
+
+      return null;
     });
 
-    try {
-      await Promise.all(allPromises);
-      refresh('channelList');
-    } catch (e) {
-      console.log('error on saving notifications:', e);
-    }
+    queue.addJob({
+      label: 'refresh channelList',
+      task: async () => {
+        refresh('channelList');
+      }
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This job is added to the queue after all the notification saving jobs. However, there's no guarantee that this job will be executed after all the previous ones have completed because jobs in the queue could be processed concurrently. If the order matters (i.e., if you need to refresh the 'channelList' only after all notifications have been saved), you might need to rethink this logic.

39:
There's a TODO comment about updating the timestamp, but I don't see any code related to that. Please ensure that this is implemented or remove the comment if it's no longer relevant.



<!-- This is an auto-generated comment by OSS CodeRabbit -->